src/main/java/Duke.java

```diff
@@ -1,10 +1,139 @@
+    }
+
+    public static void main(String[] args) throws UnknownCommandException,EmptyInputException,java.io.IOException{
+        printIntro();
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> list = new ArrayList<>();
+        int count=0;
+        //Task[] list = new Task[100];
+        String line;
+        File f = new File("data/Duke.txt");
+        if(f.exists()){
+            Scanner s = new Scanner(f);
+            while(s.hasNext()){
+                String[] temp = s.nextLine().split("\\|");
+                if(temp[0].equals("[T] ")){
+                    list.add(new ToDo(temp[2]));
+                    count++;
+                    if(temp[1].equals(" [V] ")){
+                        list.get(count-1).markAsDone();
+                    }
+                } else if(temp[0].equals("[D] ")){
+                    list.add(new Deadline(temp[2], temp[3]));
+                    count++;
+                    if(temp[1].equals(" [V] ")) {
+                        list.get(count - 1).markAsDone();
+                    }
+                } else if(temp[0].equals("[E] ")){
+                    list.add(new Event(temp[2], temp[3]));
+                    count++;
+                    if(temp[1].equals(" [V] ")) {
+                        list.get(count - 1).markAsDone();
+                    }
+                }
+            }
+        } else {
+            f.createNewFile();
+        }
+        FileWriter fw = new FileWriter(f);
+        while(true) {
+            try {
+                line = in.nextLine();
+                if (line.equals("bye")) {
+                    break;
+                } else if (line.equals("list")) {
+                    System.out.println("____________________________________________________________");
+                    System.out.println("Here are the tasks in your list:");
+                    for (int j = 0; j < count; j++) {
+                        System.out.println(j + 1 + "." + list.get(j).toString());
+                    }
+                    System.out.println("____________________________________________________________");
+                } else if (line.startsWith("done")) {
+                    String[] temp = line.split(" ");
+                    int index = Integer.parseInt(temp[1]);
+                    list.get(index - 1).markAsDone();
+                    System.out.println("Nice! I've marked this as done:");
+                    System.out.println(list.get(index - 1).toString());
+                } else if (line.startsWith("delete")) {
+                    String[] temp = line.split(" ");
+                    int index = Integer.parseInt(temp[1]);
+                    System.out.println("Noted. I've removed this task:" + System.lineSeparator() +list.get(index - 1).toString());
+                    list.remove(index-1);
+                    count--;
+                }else if (line.startsWith("todo")) {
+                    list.add(new ToDo(line.substring(4)));
```

Comment:
Avoid using magic numbers (4).
Try to redesign your logic so that it does not depends on such magic numbers